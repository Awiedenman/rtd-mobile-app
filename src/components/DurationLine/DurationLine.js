import React from 'react';

const durationLines = ({totalSteps}) => {
  const totalDuration = totalSteps.duration;
  const lineDurations = totalSteps.eachStep.map((stepDur) => {
    return (stepDur / totalDuration) * 100;
  });
  const lineContainerStyles = {
    display: 'flex',
    width: '100%'
  };
  const eachLine = lineDurations.map((line, index) => {
    const singleLineStyles = {
      display: 'grid',
      'grid-template-columns': '10px 1fr',
      'align-items': 'center'
    };
    const lineWidth = line + '%';
    const hrStyle = {
      width: lineWidth,
      border: '1px solid black'
    };
    return (
      <div 
        key={index}
        style={singleLineStyles}
      >
        <span className='dot'></span>
        <hr
          className='line-duration'
          style={hrStyle}
        />
      </div>
    );
  });
  return (
    <div
      className='line-container'
      style={lineContainerStyles}
    >
      {eachLine}
    </div>
  );
};

export default durationLines;