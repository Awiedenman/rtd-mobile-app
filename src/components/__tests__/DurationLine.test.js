import React from 'react';
import { shallow } from 'enzyme';
import DurationLine  from '../DurationLine/DurationLine';

describe('DurationLine', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <DurationLine
        totalSteps = {{
          duration: 20,
          eachStep: [20],
          stepsSum: 1
        }}
      />);

    expect(wrapper).toMatchSnapshot();
  });
});
