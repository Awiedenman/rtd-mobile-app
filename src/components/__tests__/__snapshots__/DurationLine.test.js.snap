// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DurationLine should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DurationLines
    totalSteps={
        Object {
            "duration": 20,
            "eachStep": Array [
              20,
            ],
            "stepsSum": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          style={
                    Object {
                              "align-items": "center",
                              "display": "grid",
                              "grid-template-columns": "10px 1fr",
                            }
          }
>
          <span
                    className="dot"
          />
          <hr
                    className="line-duration"
                    style={
                              Object {
                                        "border": "1px solid black",
                                        "width": "100%",
                                      }
                    }
          />
</div>,
      ],
      "className": "line-container",
      "style": Object {
        "display": "flex",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="dot"
/>,
            <hr
              className="line-duration"
              style={
                            Object {
                                          "border": "1px solid black",
                                          "width": "100%",
                                        }
              }
/>,
          ],
          "style": Object {
            "align-items": "center",
            "display": "grid",
            "grid-template-columns": "10px 1fr",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dot",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "line-duration",
              "style": Object {
                "border": "1px solid black",
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
                        Object {
                                    "align-items": "center",
                                    "display": "grid",
                                    "grid-template-columns": "10px 1fr",
                                  }
            }
>
            <span
                        className="dot"
            />
            <hr
                        className="line-duration"
                        style={
                                    Object {
                                                "border": "1px solid black",
                                                "width": "100%",
                                              }
                        }
            />
</div>,
        ],
        "className": "line-container",
        "style": Object {
          "display": "flex",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="dot"
/>,
              <hr
                className="line-duration"
                style={
                                Object {
                                                "border": "1px solid black",
                                                "width": "100%",
                                              }
                }
/>,
            ],
            "style": Object {
              "align-items": "center",
              "display": "grid",
              "grid-template-columns": "10px 1fr",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dot",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "line-duration",
                "style": Object {
                  "border": "1px solid black",
                  "width": "100%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
